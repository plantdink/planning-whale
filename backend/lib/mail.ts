import { createTransport, getTestMessageUrl } from "nodemailer";

const transport = createTransport({
  host: String(process.env.MAIL_HOST),
  port: Number(process.env.MAIL_PORT),
  auth: {
    user: process.env.MAIL_USER,
    pass: process.env.MAIL_PASS,
  },
});

function emailTemplate(text: string) {
  return `
  <div className="email" style="
    border: 1px solid black;
    padding: 20px;
    font-family: sans-serif;
    line-height: 2;
    font-size: 20px;
    ">
    <h2>Password Reset</h2>
    <p>${text}</p>
    <p>Admin team</p>
    </div>
  `;
}

export interface Envelope {
  from: string;
  to?: string[] | null;
}

export interface MailResponse {
  accepted?: string[] | null;
  rejected?: null[] | null;
  envelopeTime: number;
  messageTime: number;
  messageSize: number;
  response: string;
  envelope: Envelope;
  messageId: string;
}

export async function sendPasswordResetEmail(
  resetToken: string,
  to: string
): Promise<void> {
  const info = (await transport.sendMail({
    to,
    from: "Admin@agentfieldmanual.com",
    subject: "Your password reset token",
    html: emailTemplate(`Here is your password reset token
    <a href="${process.env.FRONTEND_URL}/reset?token=${resetToken}">Click here to reset your admin account password</a>
    `),
  })) as MailResponse;
  if (process.env.MAIL_USER.includes("ethereal.email")) {
    console.log(`Message Sent. Preview it at ${getTestMessageUrl(info)}`);
  }
}
