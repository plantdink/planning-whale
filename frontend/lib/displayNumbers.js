export function displayPercentage(number = 0) {
  const displayNumber = number / 10;
  return displayNumber;
}

export function hsDisplay(number = 0) {
  const displayNumber = number / 100;
  return displayNumber;
}
