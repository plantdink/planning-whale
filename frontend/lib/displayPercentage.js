export default function displayPercentage(number = 0) {
  const displayNumber = number / 10;
  return displayNumber;
}
