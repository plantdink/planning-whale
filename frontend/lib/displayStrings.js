export function lowerCaseFirstLetter(string) {
  const lowerCaseString = string.toLowerCase();
  const titleString = lowerCaseString.replace(
    /(^\w{1})|(\s+\w{1})/g,
    (letter) => letter.toUpperCase()
  );
  const noWhiteSpaceString = titleString.replaceAll(/\s/g, '');
  const finalLink =
    noWhiteSpaceString.charAt(0).toLowerCase() + noWhiteSpaceString.slice(1);
  return finalLink;
}

export function titleCase(string) {
  return string
    .toLowerCase()
    .split(' ')
    .map(function (word) {
      return word.charAt(0).toUpperCase() + word.slice(1);
    })
    .join(' ');
}
