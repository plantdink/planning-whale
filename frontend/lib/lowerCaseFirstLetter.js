export default function lowerCaseFirstLetter(string) {
  const lowerCaseString = string.toLowerCase();
  const titleString = lowerCaseString.replace(
    /(^\w{1})|(\s+\w{1})/g,
    (letter) => letter.toUpperCase()
  );
  const noWhiteSpaceString = titleString.replaceAll(/\s/g, '');
  const finalLink =
    noWhiteSpaceString.charAt(0).toLowerCase() + noWhiteSpaceString.slice(1);
  return finalLink;
}
