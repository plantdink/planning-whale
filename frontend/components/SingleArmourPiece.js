import { displayPercentage, humanReadableNumber } from '../lib/displayNumbers';
import HeadSEO from './HeadSEO';

export default function SingleArmourPiece({ armourPiece }) {
  return (
    <>
      <HeadSEO seoTag={armourPiece.name} />
      <>
        <div className="single-gear-item__title-bar">
          <h1 className="single-gear-item__heading">{armourPiece.name}</h1>
        </div>
        <img
          className="single-gear-item__icon-image"
          src={armourPiece.image?.image.publicUrlTransformed}
          alt={armourPiece.image?.altText}
          data-testid="itemIcon"
        />
        <>
          <div>
            {armourPiece.flavourText !== null && (
              <blockquote data-testid="flavourText">
                {armourPiece.flavourText}
              </blockquote>
            )}
            <p data-testid="itemPiece">{armourPiece.piece}</p>
            {armourPiece.notes !== '' && (
              <p data-testid="itemNotes">{armourPiece.notes}</p>
            )}
          </div>
        </>

        {/* ----------Exotic Items layout ---------- */}
        <>
          {armourPiece.__typename === 'ExoticArmourPiece' && (
            <div className="single-gear-item__content">
              <div className="single-gear-item__details">
                <div className="single-gear-item__title-bar">
                  <h2 className="single-gear-item__subheading">
                    Level 40 stats
                  </h2>
                </div>

                <div className="single-gear-item__title-bar">
                  <h3 className="single-gear-item__h3subheading">
                    Core Attribute
                  </h3>
                </div>
                <ul>
                  <li>
                    {armourPiece.maxCoreOneValueLevel40} {armourPiece.coreOne}
                  </li>
                  {armourPiece.coreTwo !== '' && (
                    <>
                      <li>
                        {armourPiece.maxCoreTwoValueLevel40}{' '}
                        {armourPiece.coreTwo}
                      </li>
                      <li>{armourPiece.coreThree}</li>
                    </>
                  )}
                </ul>

                {armourPiece.attributeOneType !== '' && (
                  <>
                    <div className="single-gear-item__title-bar">
                      <h3 className="single-gear-item__h3subheading">
                        Minor Attributes
                      </h3>
                    </div>
                    <ul>
                      <li>
                        {armourPiece.attributeOneValueLevel40}{' '}
                        {armourPiece.attributeOneType}
                      </li>
                      <li>
                        {armourPiece.attributeTwoValueLevel40}{' '}
                        {armourPiece.attributeTwoType}
                      </li>
                    </ul>
                  </>
                )}
              </div>

              {armourPiece.availableWT5 === 'Yes' && (
                <div className="single-gear-item__details">
                  <div className="single-gear-item__title-bar">
                    <h2 className="single-gear-item__subheading">
                      World Tier 5 Stats
                    </h2>
                  </div>
                  <div className="single-gear-item__title-bar">
                    <h3 className="single-gear-item__h3subheading">
                      Core Attribute
                    </h3>
                  </div>
                  <ul>
                    <li>
                      {armourPiece.maxCoreOneValueLevel30} {armourPiece.coreOne}
                    </li>
                  </ul>
                  <div className="single-gear-item__title-bar">
                    <h3 className="single-gear-item__h3subheading">
                      Minor Attributes
                    </h3>
                  </div>
                  <ul>
                    <li>
                      {armourPiece.attributeOneValueLevel30}{' '}
                      {armourPiece.attributeOneType}
                    </li>
                    <li>
                      {armourPiece.attributeTwoValueLevel30}{' '}
                      {armourPiece.attributeTwoType}
                    </li>
                  </ul>
                </div>
              )}
            </div>
          )}
        </>

        {/* -------- High end & named items Level 40 -------- */}
        <>
          {armourPiece.coreAttributeTypeOne && (
            <>
              <div
                className="single-gear-item__content"
                data-testid="singleArmourPiece"
              >
                <div className="single-gear-item__details">
                  <div className="single-gear-item__title-bar">
                    <h2 className="single-gear-item__subheading">
                      Level 40 Stats
                    </h2>
                  </div>
                  <div className="single-gear-item__title-bar">
                    <h3 className="single-gear-item__h3subheading">
                      Core Attributes
                    </h3>
                  </div>

                  <ul>
                    {armourPiece.coreAttributeTypeOne !== 'N/A' && (
                      <li data-testid="itemCoreOneType">
                        {displayPercentage(
                          armourPiece.maxCoreDamageLevel40.toString()
                        )}
                        % {armourPiece.coreAttributeTypeOne}
                      </li>
                    )}
                    {armourPiece.coreAttributeTypeTwo !== 'N/A' && (
                      <li data-testid="itemCoreTwoType">
                        {humanReadableNumber(
                          armourPiece.maxCoreArmourLevel40.toString()
                        )}{' '}
                        {armourPiece.coreAttributeTypeTwo}
                      </li>
                    )}
                    {armourPiece.coreAttributeTypeThree !== 'N/A' && (
                      <li data-testid="itemCoreThreeType">
                        {armourPiece.maxCoreSkillTierLevel40.toString()}{' '}
                        {armourPiece.coreAttributeTypeThree}
                      </li>
                    )}
                  </ul>

                  <>
                    <div className="single-gear-item__title-bar">
                      <h3 className="single-gear-item__h3subheading">
                        Minor Attributes - Offensive
                      </h3>
                    </div>
                  </>

                  {armourPiece.coreAttributeTypeOne && (
                    <>
                      <ul>
                        <li data-testid="itemOffAttrOneMax40Value">
                          {displayPercentage(
                            armourPiece.offensiveAttributeOneMaxValueLevel40.toString()
                          )}
                          % {armourPiece.offensiveAttributeOneLevel40}
                        </li>
                        <li data-testid="itemOffAttrTwoMax40Value">
                          {displayPercentage(
                            armourPiece.offensiveAttributeTwoMaxValueLevel40.toString()
                          )}
                          % {armourPiece.offensiveAttributeTwoLevel40}
                        </li>
                        <li data-testid="itemOffAttrThreeMax40Value">
                          {displayPercentage(
                            armourPiece.offensiveAttributeThreeMaxValueLevel40.toString()
                          )}
                          % {armourPiece.offensiveAttributeThreeLevel40}
                        </li>
                        <li data-testid="itemOffAttrFourMax40Value">
                          {displayPercentage(
                            armourPiece.offensiveAttributeFourMaxValueLevel40.toString()
                          )}
                          % {armourPiece.offensiveAttributeFourLevel40}
                        </li>
                      </ul>
                    </>
                  )}

                  {armourPiece.coreAttributeTypeTwo && (
                    <>
                      <div className="single-gear-item__title-bar">
                        <h3 className="single-gear-item__h3subheading">
                          Minor Attributes - Defensive
                        </h3>
                      </div>
                      <ul>
                        <li data-testid="itemDefAttrOneMax40Value">
                          {humanReadableNumber(
                            armourPiece.defensiveAttributeOneMaxValueLevel40.toString()
                          )}{' '}
                          per second {armourPiece.defensiveAttributeOneLevel40}
                        </li>
                        <li data-testid="itemDefAttrTwoMax40Value">
                          {displayPercentage(
                            armourPiece.defensiveAttributeTwoMaxValueLevel40.toString()
                          )}
                          % {armourPiece.defensiveAttributeTwoLevel40}
                        </li>
                        <li data-testid="itemDefAttrThreeMax40Value">
                          {displayPercentage(
                            armourPiece.defensiveAttributeThreeMaxValueLevel40.toString()
                          )}
                          % {armourPiece.defensiveAttributeThreeLevel40}
                        </li>
                        <li data-testid="itemDefAttrFourMax40Value">
                          {humanReadableNumber(
                            armourPiece.defensiveAttributeFourMaxValueLevel40.toString()
                          )}{' '}
                          {armourPiece.defensiveAttributeFourLevel40}
                        </li>
                      </ul>
                    </>
                  )}

                  {armourPiece.coreAttributeTypeTwo && (
                    <>
                      <div className="single-gear-item__title-bar">
                        <h3 className="single-gear-item__h3subheading">
                          Minor Attributes - Utility
                        </h3>
                      </div>
                      <ul>
                        <li data-testid="itemUtilAttrOneMax40Value">
                          {displayPercentage(
                            armourPiece.utilityAttributeOneMaxValueLevel40.toString()
                          )}
                          % {armourPiece.utilityAttributeOneLevel40}
                        </li>
                        <li data-testid="itemUtilAttrTwoMax40Value">
                          {displayPercentage(
                            armourPiece.utilityAttributeTwoMaxValueLevel40.toString()
                          )}
                          % {armourPiece.utilityAttributeTwoLevel40}
                        </li>
                        <li data-testid="itemUtilAttrThreeMax40Value">
                          {displayPercentage(
                            armourPiece.utilityAttributeThreeMaxValueLevel40.toString()
                          )}
                          % {armourPiece.utilityAttributeThreeLevel40}
                        </li>
                        <li data-testid="itemUtilAttrFourMax40Value">
                          {displayPercentage(
                            armourPiece.utilityAttributeFourMaxValueLevel40.toString()
                          )}
                          % {armourPiece.utilityAttributeFourLevel40}
                        </li>
                      </ul>
                    </>
                  )}
                </div>

                {/* ------------- World Tier 5 below ----------- */}

                {armourPiece.availableWT5 === 'Yes' && (
                  <div
                    className="single-gear-item__details"
                    data-testid="itemWT5Details"
                  >
                    <div className="single-gear-item__title-bar">
                      <h2 className="single-gear-item__subheading">
                        World Tier 5 Stats
                      </h2>
                    </div>
                    <div className="single-gear-item__title-bar">
                      <h3 className="single-gear-item__h3subheading">
                        Core Attributes
                      </h3>
                    </div>

                    <ul>
                      {armourPiece.coreAttributeTypeOne !== 'N/A' && (
                        <li>
                          {displayPercentage(
                            armourPiece.maxCoreValueDamageLevel30.toString()
                          )}
                          % {armourPiece.coreAttributeTypeOne}
                        </li>
                      )}
                      {armourPiece.coreAttributeTypeTwo !== 'N/A' && (
                        <li>
                          {humanReadableNumber(
                            armourPiece.maxCoreArmourLevel30.toString()
                          )}{' '}
                          {armourPiece.coreAttributeTypeTwo}
                        </li>
                      )}
                      {armourPiece.coreAttributeTypeThree !== 'N/A' && (
                        <li>
                          {armourPiece.maxCoreSkillTierLevel30.toString()}{' '}
                          {armourPiece.coreAttributeTypeThree}
                        </li>
                      )}
                    </ul>

                    {armourPiece.coreOne && (
                      <>
                        <div className="single-gear-item__title-bar">
                          <h3 className="single-gear-item__h3subheading">
                            Attribute One
                          </h3>
                        </div>
                        <ul>
                          <li>
                            {armourPiece.attributeOneType} -{' '}
                            {armourPiece.attributeOneValueLevel30}%
                          </li>
                        </ul>
                      </>
                    )}

                    {armourPiece.coreAttributeTypeOne && (
                      <>
                        <div className="single-gear-item__title-bar">
                          <h3 className="single-gear-item__h3subheading">
                            Minor Attributes - Offensive
                          </h3>
                        </div>
                        <ul>
                          <li>{armourPiece.attributeOneType}</li>
                          <li>{armourPiece.attributeTwoType}</li>
                        </ul>
                      </>
                    )}

                    {armourPiece.coreAttributeTypeOne && (
                      <>
                        <ul>
                          <li>
                            {displayPercentage(
                              armourPiece.offensiveAttributeOneMaxValueLevel30.toString()
                            )}
                            % {armourPiece.offensiveAttributeOneLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.offensiveAttributeTwoMaxValueLevel30.toString()
                            )}
                            % {armourPiece.offensiveAttributeTwoLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.offensiveAttributeThreeMaxValueLevel30.toString()
                            )}
                            % {armourPiece.offensiveAttributeThreeLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.offensiveAttributeFourMaxValueLevel30.toString()
                            )}
                            % {armourPiece.offensiveAttributeFourLevel40}
                          </li>
                        </ul>
                      </>
                    )}

                    {armourPiece.coreAttributeTypeTwo && (
                      <>
                        <div className="single-gear-item__title-bar">
                          <h3 className="single-gear-item__h3subheading">
                            Minor Attributes - Defensive
                          </h3>
                        </div>
                        <ul>
                          <li>
                            {humanReadableNumber(
                              armourPiece.defensiveAttributeOneMaxValueLevel30.toString()
                            )}{' '}
                            per second{' '}
                            {armourPiece.defensiveAttributeOneLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.defensiveAttributeTwoMaxValueLevel30.toString()
                            )}
                            % {armourPiece.defensiveAttributeTwoLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.defensiveAttributeThreeMaxValueLevel30.toString()
                            )}
                            % {armourPiece.defensiveAttributeThreeLevel40}
                          </li>
                          <li>
                            {humanReadableNumber(
                              armourPiece.defensiveAttributeFourMaxValueLevel30.toString()
                            )}{' '}
                            {armourPiece.defensiveAttributeFourLevel40}
                          </li>
                        </ul>
                      </>
                    )}

                    {armourPiece.coreAttributeTypeTwo && (
                      <>
                        <div className="single-gear-item__title-bar">
                          <h3 className="single-gear-item__h3subheading">
                            Minor Attributes - Utility
                          </h3>
                        </div>
                        <ul>
                          <li>
                            {displayPercentage(
                              armourPiece.utilityAttributeOneMaxValueLevel30.toString()
                            )}
                            % {armourPiece.utilityAttributeOneLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.utilityAttributeTwoMaxValueLevel30.toString()
                            )}
                            % {armourPiece.utilityAttributeTwoLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.utilityAttributeThreeMaxValueLevel30.toString()
                            )}
                            % {armourPiece.utilityAttributeThreeLevel40}
                          </li>
                          <li>
                            {displayPercentage(
                              armourPiece.utilityAttributeFourMaxValueLevel30.toString()
                            )}
                            % {armourPiece.utilityAttributeFourLevel40}
                          </li>
                        </ul>
                      </>
                    )}
                  </div>
                )}
              </div>

              {(armourPiece.uniqueAttribute || armourPiece.modSlots > 0) && (
                <div className="single-gear-item__sub-content">
                  <div className="single-gear-item__details">
                    {armourPiece.uniqueAttribute && (
                      <>
                        <div className="single-gear-item__title-bar">
                          <h3 className="single-gear-item__h3subheading">
                            Item Unique Attribute
                          </h3>
                        </div>
                        <ul>
                          <li>{armourPiece.uniqueAttribute}</li>
                        </ul>
                      </>
                    )}

                    {armourPiece.modSlots > 0 && (
                      <>
                        <div className="single-gear-item__title-bar">
                          <h3 className="single-gear-item__h3subheading">
                            Number of Mod Slots
                          </h3>
                        </div>
                        <ul>
                          <li>{armourPiece.modSlots}</li>
                        </ul>
                      </>
                    )}
                  </div>
                </div>
              )}
            </>
          )}
        </>
      </>
    </>
  );
}
