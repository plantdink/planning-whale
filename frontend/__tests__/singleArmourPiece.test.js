import { render, screen, cleanup } from '@testing-library/react';
import SingleArmourPiece from '../components/SingleArmourPiece';
import { fakeArmourPiece, fakeNamedArmourPiece } from '../lib/testUtils';
import { displayPercentage, humanReadableNumber } from '../lib/displayNumbers';

const armourPiece = fakeArmourPiece();

afterEach(cleanup);

describe('<SingleArmourPiece />', () => {
  it('renders correctly for a single armour piece', async () => {
    const { container, debug } = render(
      <SingleArmourPiece armourPiece={armourPiece} />
    );

    await screen.findByTestId('singleArmourPiece');
    expect(container).toMatchSnapshot();

    const testArmourPieceTitle = screen.getByText(armourPiece.name);
    expect(testArmourPieceTitle).toBeInTheDocument();

    const testArmourPieceIcon = await screen.findByTestId('itemIcon');
    expect(testArmourPieceIcon).toBeInTheDocument();

    const testFlavourText = await screen.findByTestId('flavourText');
    if (armourPiece.flavourText !== null)
      expect(testFlavourText).toBeInTheDocument();
    expect(testFlavourText).toHaveTextContent(armourPiece.flavourText);

    const testPiece = await screen.findByTestId('itemPiece');
    if (armourPiece.piece !== '') expect(testPiece).toBeInTheDocument();
    expect(testPiece).toHaveTextContent(armourPiece.piece);

    const testNotes = await screen.findByTestId('itemNotes');
    if (armourPiece.notes !== '') expect(testNotes).toBeInTheDocument();
    expect(testNotes).toHaveTextContent(armourPiece.notes);
  });
});
