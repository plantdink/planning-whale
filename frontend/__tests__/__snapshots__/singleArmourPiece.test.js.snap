// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SingleArmourPiece /> renders correctly for a single armour piece 1`] = `
<div>
  <div
    class="single-gear-item__title-bar"
  >
    <h1
      class="single-gear-item__heading"
    >
      Nightwatcher
    </h1>
  </div>
  <img
    alt="Mask icon"
    class="single-gear-item__icon-image"
    data-testid="itemIcon"
    src="https://res.cloudinary.com/dbm1h5w02/image/upload/v1647498165/agent-field-manual/6232d3b4ec80b2199ff2c03a.png"
  />
  <div>
    <blockquote
      data-testid="flavourText"
    >
      Through glass and steel; looking at you; a view so cruel.
    </blockquote>
    <p
      data-testid="itemPiece"
    >
      Mask
    </p>
    <p
      data-testid="itemNotes"
    >
      This mask has been manufactured by Gila Guard and comes with Armour as the primary Core attribute. It has one fixed minor attribute - 100% Scanner Pulse Haste - and one random minor attribute which can be any of the stats listed below. It also has one gear mod slot.
    </p>
  </div>
  <div
    class="single-gear-item__content"
    data-testid="singleArmourPiece"
  >
    <div
      class="single-gear-item__details"
    >
      <div
        class="single-gear-item__title-bar"
      >
        <h2
          class="single-gear-item__subheading"
        >
          Level 40 Stats
        </h2>
      </div>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Core Attributes
        </h3>
      </div>
      <ul>
        <li
          data-testid="itemCoreTwoType"
        >
          170,000
           
          Armour
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Minor Attributes - Offensive
        </h3>
      </div>
      <ul>
        <li
          data-testid="itemOffAttrOneMax40Value"
        >
          6
          % 
          Critical Hit Chance
        </li>
        <li
          data-testid="itemOffAttrTwoMax40Value"
        >
          15
          % 
          Critical Hit Damage
        </li>
        <li
          data-testid="itemOffAttrThreeMax40Value"
        >
          10
          % 
          Headshot Damage
        </li>
        <li
          data-testid="itemOffAttrFourMax40Value"
        >
          8
          % 
          Weapon Handling
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Minor Attributes - Defensive
        </h3>
      </div>
      <ul>
        <li
          data-testid="itemDefAttrOneMax40Value"
        >
          4,925
           
          per second 
          Armour Regeneration
        </li>
        <li
          data-testid="itemDefAttrTwoMax40Value"
        >
          10
          % 
          Explosive Resistance
        </li>
        <li
          data-testid="itemDefAttrThreeMax40Value"
        >
          10
          % 
          Hazard Protection
        </li>
        <li
          data-testid="itemDefAttrFourMax40Value"
        >
          18,935
           
          Health
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Minor Attributes - Utility
        </h3>
      </div>
      <ul>
        <li
          data-testid="itemUtilAttrOneMax40Value"
        >
          10
          % 
          Skill Damage
        </li>
        <li
          data-testid="itemUtilAttrTwoMax40Value"
        >
          12
          % 
          Skill Haste
        </li>
        <li
          data-testid="itemUtilAttrThreeMax40Value"
        >
          20
          % 
          Skill Repair
        </li>
        <li
          data-testid="itemUtilAttrFourMax40Value"
        >
          10
          % 
          Status Effects
        </li>
      </ul>
    </div>
    <div
      class="single-gear-item__details"
      data-testid="itemWT5Details"
    >
      <div
        class="single-gear-item__title-bar"
      >
        <h2
          class="single-gear-item__subheading"
        >
          World Tier 5 Stats
        </h2>
      </div>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Core Attributes
        </h3>
      </div>
      <ul>
        <li>
          42,458
           
          Armour
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Minor Attributes - Offensive
        </h3>
      </div>
      <ul>
        <li />
        <li />
      </ul>
      <ul>
        <li>
          5.1
          % 
          Critical Hit Chance
        </li>
        <li>
          7.9
          % 
          Critical Hit Damage
        </li>
        <li>
          6.9
          % 
          Headshot Damage
        </li>
        <li>
          7
          % 
          Weapon Handling
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Minor Attributes - Defensive
        </h3>
      </div>
      <ul>
        <li>
          1,223
           
          per second
           
          Armour Regeneration
        </li>
        <li>
          6.6
          % 
          Explosive Resistance
        </li>
        <li>
          6.6
          % 
          Hazard Protection
        </li>
        <li>
          4,695
           
          Health
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Minor Attributes - Utility
        </h3>
      </div>
      <ul>
        <li>
          6.9
          % 
          Skill Damage
        </li>
        <li>
          7.9
          % 
          Skill Haste
        </li>
        <li>
          12.1
          % 
          Skill Repair
        </li>
        <li>
          6.6
          % 
          Status Effects
        </li>
      </ul>
    </div>
  </div>
  <div
    class="single-gear-item__sub-content"
  >
    <div
      class="single-gear-item__details"
    >
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Item Unique Attribute
        </h3>
      </div>
      <ul>
        <li>
          100% Pulse Haste
        </li>
      </ul>
      <div
        class="single-gear-item__title-bar"
      >
        <h3
          class="single-gear-item__h3subheading"
        >
          Number of Mod Slots
        </h3>
      </div>
      <ul>
        <li>
          1
        </li>
      </ul>
    </div>
  </div>
</div>
`;
